connector.go

```diff
@@ -202,6 +192,24 @@ func (wrb *WapiRequestBuilder) BuildBody(obj IBObject) []byte {
 		return nil
 	}
 
+	if t == GET && len(eaSearch) > 0 {
+		payload := make(map[string]string)
+		json.Unmarshal(jsonStr, &payload)
+		for k, v := range eaSearch {
+			str, ok := v.(string)
+			if !ok {
+				log.Printf("Cannot marshal EA Search attribute for '%s'\n", k)
+			} else {
+				payload["*"+k] = str
+			}
+		}
+		jsonStr, err = json.Marshal(payload)
+		if err != nil {
+			log.Printf("Cannot marshal EA's in the payload: '%s'", payload)
+			return nil
+		}
+	}
+
```

Comment:
I assume this is just a WIP....

Instead of doing the Unmashal -> add EASearch -> Marshall, you should consider doing the following:
1) Define a new type for EASearch field
2) Define a custom Marshaller for that
3) Add EASearch field to be part of the main object (instead of being in IBBase)

Then the original BuildBody() method should be able to handle it.